case_studies/CARD/Bug-Drug Code.R

```diff
@@ -0,0 +1,148 @@
+# config.R
+url <- "https://card.mcmaster.ca/download/0/broadstreet-v3.3.0.tar.bz2"
+destfile <- "broadstreet-v3.3.0.tar.bz2"
+
+# Download the file
+download.file(url, destfile)
+
+#Extract the file
+if (!require("R.utils")) {
+  install.packages("R.utils")
+  library(R.utils)
+}
+
+
+# Extract the tar file
+untar("broadstreet-v3.3.0.tar", exdir = "CARD_data")
+
+
+# Map CARD Short Name
+# Install and Load dplyr
+if (!require("dplyr")) {
+  install.packages("dplyr")
+  library(dplyr)
+} else {
+  library(dplyr)
+}
+
+#  Read the required files
+aro_index <- read.delim("CARD_data/aro_index.tsv", sep = "\t", header = TRUE)
+antibiotics_data <- read.delim("CARD_data/shortname_antibiotics.tsv", sep = "\t", header = TRUE)
+pathogens_data <- read.delim("CARD_data/shortname_pathogens.tsv", sep = "\t", header = TRUE)
+
+
+# Extract pathogen, gene, drug, and include Protein.Accession from 'CARD.Short.Name'
+aro_index_clean <- aro_index %>%
+  mutate(
+    pathogen = sapply(strsplit(CARD.Short.Name, "_"), `[`, 1),  # Extract pathogen
+    gene = sapply(strsplit(CARD.Short.Name, "_"), `[`, 2),      # Extract gene
+    drug = ifelse(sapply(strsplit(CARD.Short.Name, "_"), length) == 3,   # Extract drug
+                  sapply(strsplit(CARD.Short.Name, "_"), `[`, 3), NA),
+    Protein.Accession = Protein.Accession  # Include the Protein.Accession column
+  )
+
+# Merge aro_index_clean with the antibiotics_data and pathogens_data
+# For merging with antibiotics_data
+merged_data_antibiotics <- left_join(aro_index_clean, antibiotics_data,
+                                     by = c("drug" = "AAC.Abbreviation"))
+
+# For merging with pathogens_data
+merged_data_pathogens <- left_join(merged_data_antibiotics, pathogens_data,
+                                   by = c("pathogen" = "Abbreviation"))
+
+# View the resulting merged data
+head(merged_data_pathogens)
+
+
+#filter out rows where pathogen is empty
+cleaned_data <- merged_data_pathogens %>%
+  distinct() %>%
+  filter(!is.na(Pathogen)) # Use 'Pathogen' instead of 'pathogen'
+View(cleaned_data)
+
+# Group by Pathogen, Gene, Drug, and Protein.Accession, then summarize Antibiotic information
+summarized_data <- cleaned_data %>%
+  group_by(Pathogen = Pathogen, Gene = gene, Drug = drug, Protein_Accession = Protein.Accession) %>%
+  summarize(Antibiotic_Info = paste(unique(Molecule), collapse = ", ")) %>%
+  arrange(Pathogen, Gene, Drug, Protein_Accession)
```

Comment:
there's a Protein_Accession here, and Protein.Accession in L65. Finally, there's protein_accession (no_caps) on L95. Maybe streamline this a bit?